engine/userguide/storagedriver/btrfs-driver.md

```diff
@@ -6,216 +6,98 @@ title: Docker and Btrfs in practice
-development with a focus on stability, this is now becoming more of a reality.
+`btrfs` is supported if you meet the following prerequisites:
 
-As far as Docker on the Linux platform goes, many people see the `btrfs`
-storage driver as a potential long-term replacement for the `devicemapper`
-storage driver. However, at the time of writing, the `devicemapper` storage
-driver should be considered safer, more stable, and more *production ready*.
-You should only consider the `btrfs` driver for production deployments if you
-understand it well and have existing experience with Btrfs.
+- **Docker CE**: For Docker CE, `btrfs` is only recommended on Ubuntu or Debian.
 
-## Image layering and sharing with Btrfs
+- **Docker EE**: For Docker EE and CS-Engine, `btrfs` is only supported on SLES.
+  See the
+  [Product compatibility matrix](https://success.docker.com/Policies/Compatibility_Matrix)
+  for all supported configurations for commercially-supported Docker.
 
-Docker leverages Btrfs *subvolumes* and *snapshots* for managing the on-disk
-components of image and container layers.  Btrfs subvolumes look and feel like
-a normal Unix filesystem. As such, they can have their own internal directory
-structure that hooks into the wider Unix filesystem.
+- Changing the storage driver will make any containers you have already
+  created inaccessible on the local system. Use `docker save` to save containers,
+  and push existing images to Docker Hub or a private repository, so that you
+  not need to re-create them later.
 
-Subvolumes are natively copy-on-write and have space allocated to them
-on-demand from an underlying storage pool. They can also be nested and snapped.
- The diagram below shows 4 subvolumes. 'Subvolume 2' and 'Subvolume 3' are
-nested, whereas 'Subvolume 4' shows its own internal directory tree.
+- `btrfs` requires a dedicated block storage device such as a physical disk. This
+  block device must be formatted for Btrfs and mounted into `/var/lib/docker/`.
+  The configuration instructions below walk you through this procedure. By
+  default, the SLES `/` filesystem is formatted with BTRFS, so for SLES, you do
+  not need to use a separate block device, but you can choose to do so for
+  performance reasons.
 
-![](images/btfs_subvolume.jpg)
+- `btrfs` support must exist in your kernel. To check this, run the following
+  command:
 
-Snapshots are a point-in-time read-write copy of an entire subvolume. They
-exist directly below the subvolume they were created from. You can create
-snapshots of snapshots as shown in the diagram below.
+  ```bash
+  $ sudo cat /proc/filesystems | grep btrfs
 
-![](images/btfs_snapshots.jpg)
+  btrfs
+  ```
 
-Btrfs allocates space to subvolumes and snapshots on demand from an underlying
-pool of storage. The unit of allocation is referred to as a *chunk*, and
-*chunks* are normally ~1GB in size.
+  If support is not included, install the `btrfsprogs` package (SLES) or
```

Comment:
This doesn't sound right. Can you have `btrfs` missing from `/proc/filesystems` but still have the module loaded `lsmod | grep btrfs`. My understanding is `/proc/filesystems` returns the support filesystems in the kernel and if it is missing there you need a different kernel. The progs and tools just provide utilities for operating with btrfs since btrfs operations require ioctls which are not supports by the standard fs tools, but they are not required for mounting and using btrfs.